dkist/dataset/loader.py

```diff
@@ -0,0 +1,73 @@
+import importlib.resources as importlib_resources
+from pathlib import Path
+from functools import singledispatch
+
+from jsonschema.exceptions import ValidationError
+
+import asdf
+
+
+@singledispatch
+def load_dataset():
+    """Function to load a Dataset from either an asdf file path or a directory."""
```

Comment:
This should raise an unknown type error as I think it gets called for any unknown types.